### PR TITLE
CPRI-4110: Fetch AssetPlan on Import and Create/Edit AssetRequest Screen

### PR DESCRIPTION
**Pull request by propattern**

### CODE REVIEW

File: src/services/assetPlan/__test__/getAssetPlan.test.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+  it('returns the expected Briefcase', async () => {
+    const briefcase = await getAssetPlan(config, planURN)
+    expect(briefcase).to.deep.equal({
+      _patchData:
+        { id: 'someId', etags: 'etagsValue' },
+      mappedData: { mappedDataField: 'mappedDataValue' }
+    })
+  })
+})
```
Comment (rishab-mehra) [inline src/services/assetPlan/__test__/getAssetPlan.test.js]: Nice test coverage 👍

File: src/services/register/mappers/register.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import { mapRegistry } from '../../registry'
```
Comment (devu) [inline src/services/register/mappers/register.js]: can you please clarify what is registry and register? This part confusing me a lot :)

File: src/services/assetPlan/__test__/getAssetPlan.test.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+const mdsGetStub = sinon.stub().returns(mdsGetData)
+const assetPlanGraphStub = sinon.stub()
+const mapAssetPlanStub = sinon.stub().returns({ mappedDataField: 'mappedDataValue' })
+
+// eslint-disable-next-line no-underscore-dangle
+rewireAPI.__Rewire__('mdsGet', mdsGetStub)
+// eslint-disable-next-line no-underscore-dangle
+rewireAPI.__Rewire__('assetPlanGraph', assetPlanGraphStub)
+// eslint-disable-next-line no-underscore-dangle
+rewireAPI.__Rewire__('mapAssetPlan', mapAssetPlanStub)
```
Comment (rishab-mehra) [inline src/services/assetPlan/__test__/getAssetPlan.test.js]: really @propattern  rewire is very nice mocking/stubbing library as compared to proxyquire and code looks neat and clean

File: src/services/index.js [hunk 4]
```diff
@@ -7,10 +7,10 @@ export { getRightsAssetCategoryTaxonomy } from './taxonomy/getRightsAssetCategor
...
 export { getRightsReleaseTypeTaxonomy } from './taxonomy/getRightsReleaseTypeTaxonomy'
 export { getAllTaxonomies } from './taxonomy/getAllTaxonomies'
-export { getProductId } from './getProductId'
 export { getDuplicateRequestId } from './getDuplicateRequestId'
 export { getRightsClassTaxonomy } from './taxonomy/getRightsClassTaxonomy'
 export { getRightsHolderTaxonomy } from './taxonomy/getRightsHolderTaxonomy'
 export { getRightsSourceCollectionTaxonomy } from './taxonomy/getRightsSourceCollectionTaxonomy'
 export { getRoleTaxonomy } from './taxonomy/getRoleTaxonomy'
 export { getSelectionTypeTaxonomy } from './taxonomy/getSelectionTypeTaxonomy'
+export * from './assetPlan'
```
Comment (devu) [inline src/services/index.js]: if we trying to be specific and consistent ;)
export { getAssetPlan } from './assetPlan'

File: src/services/assetPlan/graph.js [hunk 5]
```diff
@@ -0,0 +1,41 @@
+import { WholeObject, Ref, RefCollection } from '@pearson-content-components/mds-client'
+
+export const assetPlanGraph = {
+  id: String,
```
Comment (devu) [inline src/services/assetPlan/graph.js]: This started to look like TypeScript or Java to me ;)

File: src/services/assetPlan/getAssetPlan.js [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+  const getAssetPlanOptions = {
+    headers: {
+      Accept: 'application/ld+json; version=1',
+      Prefer: 'etag=header, '
+      + 'return=representation;'
+      + 'embedLevel=2;predicate="usesRegister,inRegistry,identifiedBy";'
+      + 'omit="hasRegister"'
+    }
+  }
+  const { etags, data } = await mdsGet(assetPlanGraph, config, urn, getAssetPlanOptions)
```
Comment (rishab-mehra) [inline src/services/assetPlan/getAssetPlan.js]: @propattern I just want to know, suppose if there is connectivity issue from mds side or we are not getting any response. so what error do we get ? I mean to  say do we need to handle the failure response from mds side ?.

Comment (propattern) [reply src/services/assetPlan/getAssetPlan.js]: I am not sure if we are meant to be doing anything if there is no response.  A user will get a 500 error of sort and then we can later do what we want.

MDS is a core part of the application, without it Blueprint is not meant to be function.  For example if your computer hard disk gets pulled, it is not you computer's job to fix itself.  So for now we don't have to do anything.

Hopefully at a later date we will do something at service later which will then throw all these errors up and then do something about it.

File: src/services/register/mappers/register.js (hunk 2 above)
Comment (propattern) [reply src/services/register/mappers/register.js]: So the structure is:
AssetPlan -> Register -> Registry

The form data that we capture on create/edit of plan page store the value in Distributable & Register.

The Register gets linked to existing Registry []

The Registry (which represent a system, Alfresco in our case) contains information and links (urls) about that system and we need those urls to work out thumbnails and also pass it to Asset Picker.

So:
Asset Plan [Distributable] such as Book
Register: A link or entity that says in which system the Plan (Book) is residing
Register: A resource that represent a system i.e. Alfresco

File: src/services/index.js (hunk 4 above)
Comment (propattern) [reply src/services/index.js]: Cool, pushed a quick fix for this :-)

**Comment by devu** [comment]
Cool, merging this one

**Pull request merged**